### PR TITLE
[Doc]Fix typo and language issues in the faq.md file

### PR DESCRIPTION
**Pull request by Jennifer88huang**
### Motivation
Fix typo and language issues in the faq.md file. I'm trying to make it clear, concise and logical.

### Modifications
1. Refine typos;
2. Refine from the language perspective, make it easy to read and understand;
3. Refine some answers, make it clear and correct;
4. Try to re-categorize question types (not finished yet).

### CODE REVIEW

**Comment by Jennifer88huang** [comment]
@sijie @jiazhai could you please help review when you're available?
and please refine the question in Line 187 (How can I prevent an inactive topic to be deleted under any circumstance? I do not want to set time or space limit for a certain namespace.)

File: faq.md [hunk 1]
```diff
@@ -112,162 +111,135 @@ It’s a component that was introduced recently. Essentially it’s a stateless
...
 
-### Is there a profiling option in Pulsar, so that we can breakdown the time costed in every stage? For instance, message A stay in queue 1ms, bk writing time 2ms(interval between sending to bk and receiving ack from bk) and so on.
-There are latency stats at different stages. In the client (eg: reported every 1min in info logs). 
-In the broker: accessible through the broker metrics, and finally in bookies where there are several different latency metrics. 
+### How can I prevent an inactive topic to be deleted under any circumstance? I do not want to set time or space limit for a certain namespace.
+Currently, there is not an option for “infinite” (though it sounds a good idea! Maybe we could use `-1` for that). The only option is to use INT_MAX for `retentionTimeInMinutes` and LONG_MAX for `retentionSizeInMB`. It’s not “infinite”, yet 4085 years of retention is probably be enough!
+<!--@Sijie: could you please refine this answer when you read here?-->
```
Comment (sijie) [inline faq.md]: @Jennifer88huang 

The answer is out of date.

Pulsar supports infinite retention now. See section: http://pulsar.apache.org/docs/en/cookbooks-retention-expiry/#retention-policies

File: faq.md [hunk 2]
```diff
@@ -112,162 +111,135 @@ It’s a component that was introduced recently. Essentially it’s a stateless
...
+A third option is to check the topic stats. This is a tiny bit involved, because it requires the admin client (or using REST) to get the stats for the topic and checking the "backlog". If the backlog is 0, it means you've hit the end.
 
-### Is there a profiling option in Pulsar, so that we can breakdown the time costed in every stage? For instance, message A stay in queue 1ms, bk writing time 2ms(interval between sending to bk and receiving ack from bk) and so on.
-There are latency stats at different stages. In the client (eg: reported every 1min in info logs). 
-In the broker: accessible through the broker metrics, and finally in bookies where there are several different latency metrics. 
+### How can I prevent an inactive topic to be deleted under any circumstance? I do not want to set time or space limit for a certain namespace.
+Currently, there is not an option for “infinite” (though it sounds a good idea! Maybe we could use `-1` for that). The only option is to use INT_MAX for `retentionTimeInMinutes` and LONG_MAX for `retentionSizeInMB`. It’s not “infinite”, yet 4085 years of retention is probably be enough!
```
Comment (Jennifer88huang) [inline faq.md]:
```suggestion
You can set *infinite* retention time or size, by setting `-1` for either time or
size retention.
For more details, see [Pulsar retention policy](http://pulsar.apache.org/docs/en/cookbooks-retention-expiry/#retention-policies).
```

File: faq.md (hunk 1 above)
Comment (Jennifer88huang) [inline faq.md]:
```suggestion

```

File: faq.md [hunk 3]
```diff
@@ -112,162 +111,136 @@ It’s a component that was introduced recently. Essentially it’s a stateless
...
-
-### Is there a way of having both authentication and the Pulsar dashboard working at same time?
-The key is that with authorization, the stats collector needs to access the APIs that require the credentials. That’s not a problem for stats collected through Prometheus but it is for the “Pulsar dashboard” which is where the per-topic stats are shown. I think that should be quite easy to fix.
+Yes, the “proxy” is an additional component to deploy a stateless proxy frontend that can be exposed through a load balancer, and that doesn’t require direct connectivity to the actual brokers. You do not need to use the proxy within Kubernetes cluster. In some cases, it’s simpler to expose the brokers through `NodePort` or `ClusterIp` for other producer/consumers outsides.
 
 ### How can I know when I've reached the end of the stream during replay?
```
Comment (ivankelly) [inline faq.md]: This FAQ needs to be updated. There is a method to know now.

File: faq.md [hunk 4]
```diff
@@ -112,162 +111,136 @@ It’s a component that was introduced recently. Essentially it’s a stateless
...
+There are latency stats at different stages: client, bookie and broker. 
+For example, the client reports every minute in info logs. 
+The broker is accessible through the broker metrics, and bookies store several different latency metrics. 
 
-### How can I have multiple readers that each get all the messages from a topic from the beginning concurrently? I.e., no round-robin, no exclusivity
-you can create reader with `MessageId.earliest`
+In broker, there’s just the write latency on Bookkeeper, because there is no other queuing involved in the write path.
 
+### How can multiple readers get all the messages from a topic from the beginning concurrently? I.e., no round-robin, no exclusivity.
+You can create reader with `MessageId.earliest`.
```
Comment (ivankelly) [inline faq.md]: This FAQ makes little sense. The answer is in the question.

File: faq.md [hunk 5]
```diff
@@ -112,162 +111,136 @@ It’s a component that was introduced recently. Essentially it’s a stateless
...
 
 ### Does broker validate if a property exists or not when producer/consumer connects ?
-yes, broker performs auth&auth while creating producer/consumer and this information presents under namespace policies.. so, if auth is enabled then broker does validation
+Yes, broker performs auth&auth while creating producer/consumer and this information presents under namespace policies. So, if auth is enabled, and then broker performs validation.
```
Comment (ivankelly) [inline faq.md]: "Property" should be "tenant" now. 

Answer is unclear. auth & auth is authentication and authorization, but which one needs to be enabled for validation.

File: faq.md [hunk 6]
```diff
@@ -112,162 +111,136 @@ It’s a component that was introduced recently. Essentially it’s a stateless
...
+
+### For subscribing to a large number of topics like that, would I need to call `subscribe` for each one individually, or is there some sort of wildcard capability?
```
Comment (ivankelly) [inline faq.md]: This is regex subscription. it exists now.

File: faq.md (hunk 5 above)
Comment (Jennifer88huang) [reply faq.md]: @sijie @jiazhai could you please confirm which auth needs to be enabled?

File: faq.md [hunk 7]
```diff
@@ -112,162 +111,136 @@ It’s a component that was introduced recently. Essentially it’s a stateless
...
+The broker is accessible through the broker metrics, and bookies store several different latency metrics. 
 
-### How can I have multiple readers that each get all the messages from a topic from the beginning concurrently? I.e., no round-robin, no exclusivity
-you can create reader with `MessageId.earliest`
+In broker, there’s just the write latency on Bookkeeper, because there is no other queuing involved in the write path.
 
+### How can multiple readers get all the messages from a topic from the beginning concurrently? I.e., no round-robin, no exclusivity.
+You can create reader with `MessageId.earliest`.
 
 ### Does broker validate if a property exists or not when producer/consumer connects ?
```
Comment (Jennifer88huang) [inline faq.md]:
```suggestion
### Does broker validate if a tenant exists or not when producer/consumer connects ?
```

File: faq.md [hunk 8]
```diff
@@ -2,108 +2,107 @@
...
-2. Topics are matched to bundles by hashing on the name
-3. Effectively, a bundle is a hash-range where topics falls into
-4. Initially the default is to have 4 “bundles” for a namespace
-5. When the traffic increases on a given bundle, it will be split in 2 and reassigned to a different broker
-6. There are some adjustable thresholds that can be used to control when the split happens, based on number of topics/partitions, messages in/out, bytes in/out, etc..
-7. It’s also possible to specify a higher number of bundles when creating a namepsac
-8. There are the load-manager threshold that control when a broker should offload some of the bundles to other brokers
+### Where do I look into to tweak load balancing ?
+There are a few parameters to look at :
+1. The topic assignments to brokers are done in terms of “bundles”, that is in group of topic.
```
Comment (Jennifer88huang) [inline faq.md]:
```suggestion
1. The topic assignments to brokers are done in terms of “bundles”, which are groups of topics.
```

File: faq.md [hunk 9]
```diff
@@ -2,108 +2,107 @@
...
+2. Topics are matched to bundles by hashing on the name.
+3. A bundle is a hash-range where topics fall into.
```
Comment (Jennifer88huang) [inline faq.md]:
```suggestion
3. A bundle is a hash-range into which topics fall.
```

File: faq.md (hunk 3 above)
Comment (Jennifer88huang) [reply faq.md]: What's the method? how to conduct it?

File: faq.md (hunk 4 above)
Comment (Jennifer88huang) [reply faq.md]: so shall we just remove this FAQ?

File: faq.md [hunk 10]
```diff
@@ -2,108 +2,107 @@
...
 
 #### Broker level isolation
-There is the option to isolate certain tenants/namespaces to a particular set of broker. Typically the reason for using that was to experiment with different configurations, debugging and quickly react to unexpected situations.
+There is an option to isolate certain tenants/namespaces to a particular set of brokers. Typically, you use the option when you wish to experiment with different configurations, debug or quickly react to unexpected situations.
 
-For example, a particular user might be triggering a bad behavior in the broker that can impact performance for other tenants.
+For example, a particular user might trigger a bad behavior in the broker that impacts performance for other tenants.
 
-In this case, the particular user can be "isolated" a subset of brokers that will not serve any other traffic, until a proper fix that correctly handles the condition can be deployed.
+In this case, the particular user is "isolated" to a subset of brokers that do not serve any other traffic, until a proper fix that correctly handles the condition is deployed.
```
Comment (Jennifer88huang) [inline faq.md]:
```suggestion
In this case, the particular user can be "isolated" to a subset of brokers that do not serve any other traffic, until a proper fix that correctly handles the condition is deployed.
```

File: faq.md [hunk 11]
```diff
@@ -112,162 +111,136 @@ It’s a component that was introduced recently. Essentially it’s a stateless
...
+So, when Pulsar cluster recognizes the new formats in a release, it starts using new formats in the next release. 
 
-So, one release will understand the new format while the next one will actually start using it.
+It is allowed to downgrade a running cluster to a previous version, in case any server problem is identified in production.
 
-### Since Pulsar has configurable retention per namespace, can I set a "forever" value, ie., always keep all data in the namespaces?
-So, retention applies to "consumed" messages. Ones, for which the consumer has already acknowledged the processing. By default, retention is 0, so it means data is deleted as soon as all consumers acknowledge. You can set retention to delay the retention.
+### Since Pulsar has configurable retention per namespace, can I set a "forever" value, and keep all data in the namespaces forever?
+Retention applies to "consumed" messages, for which the consumer has already acknowledged the processing. By default, retention is set to 0, which means data is deleted as soon as all consumers acknowledge. You can set retention to delay the retention.
```
Comment (Jennifer88huang) [inline faq.md]:
```suggestion
You can set *infinite* retention time or size, by setting `-1` for either time or size retention.
For more details, see [Pulsar retention policy](http://pulsar.apache.org/docs/en/cookbooks-retention-expiry/#retention-policies).
```

File: faq.md [hunk 12]
```diff
@@ -112,162 +111,136 @@ It’s a component that was introduced recently. Essentially it’s a stateless
...
 
-That also means, that data is kept forever, by default, if the consumers are not acknowledging.
+It also means that data is kept forever, by default, if the consumers are not acknowledging.
```
Comment (Jennifer88huang) [inline faq.md]:
```suggestion

```

File: faq.md [hunk 13]
```diff
@@ -2,108 +2,107 @@
...
+In Pulsar, "namespace" is the administrative unit: you can configure most options on a namespace, and the configuration is applied on the topics contained on the namespace. It is convenient to configure settings and operations on a group of topics rather than doing it once per topic.
 
-In general, the pattern is to use a namespace for each user application. So a single user/tenant, can create multiple namespaces to manage its own applications.
+In general, the pattern is to use a namespace for each user application. So a single user/tenant can create multiple namespaces to manage its own applications.
 
-When it comes to topics, we need a way to assign topics to brokers, control the load and move them if a broker becomes overloaded. Rather that doing this operations per each single topic (ownership, load-monitoring, assigning), we do it in bundles, or "groups of topics".
+Concerning topics, topics are assigned to brokers, control the load and move them if a broker becomes overloaded. Rather than doing these operations per each single topic (ownership, load-monitoring, assigning), we do it in bundles, or "groups of topics".
```
[the remark this answers is not in the document]
Comment (Jennifer88huang) [reply faq.md]: so shall we remove it or refine it?

**Comment by sijie** [comment]
@Jennifer88huang I am closing this file. Since this doesn't seem to be needed any more.

**Comment by Jennifer88huang** [comment]
Ok.

**Pull request closed without merging**